### PR TITLE
SetupAuthoring for Roslyn localization

### PR DESCRIPTION
**Pull request by Cosifne**
[Issue](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/723558)
So this PR will enable 
ExpressionEvaluatorPackage.vsix, Roslyn.VisualStudio.InteractiveComponents.vsix and Roslyn.VisualStudio.Setup.vsix contains the Satellite Dlls.

As for Microsoft.CodeAnalysis.Compilers.vsix, I checked my current VS2017 directory under C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Roslyn and there is no Resources file in there, so I guess it won't need it. @tmeschter for double check.

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@Cosifne I know there was some discussion about whether we should have separate localization packages rather than doing this approach; is that something we rejected?

**Comment by Cosifne** [comment]
> @Cosifne I know there was some discussion about whether we should have separate localization packages rather than doing this approach; is that something we rejected?

Yes, previously Tom and I have done some work [here](https://github.com/dotnet/roslyn/pull/31247), it did what you described (create a different packages contain the resources files and then build and sign them) I have talked to Tom and this is no what we want, then I switch to this approach

**Comment by jasonmalinowski** [comment]
@Cosifne Awesome, sounds good!

**Comment by tmeschter** [comment]
@Cosifne @jasonmalinowski We should try get back to a place where each language is in it's own package. However, in the interests of getting loc working for Preview 2 we should just go with this approach.

**Comment by Cosifne** [comment]
> 
> 
> @Cosifne We do need resources for Microsoft.CodeAnalysis.Compilers.vsix. Or to be more exact, we need the satellite assemblies to install under MSBuild<version>\Bin\Roslyn--I assume Microsoft.CodeAnalysis.Compilers.vsix is the way to do that.
> 
> If you don't see any satellite assemblies under there now, it may be because you don't have any language packs installed.

Sure, I can add these property group to Compiler packages

**Comment by Cosifne** [comment]
tag @tmeschter to review it

**Comment by jasonmalinowski** [comment]
> However, in the interests of getting loc working for Preview 2 we should just go with this approach for now.

What is tracking fixing this?

File: src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj [hunk 1]
```diff
@@ -70,8 +70,68 @@
         <NGenPriorityString Condition="'%(CompilerArtifact.NGenPriority)' != ''"> vs.file.ngenPriority=%(CompilerArtifact.NGenPriority)</NGenPriorityString>
         <NGenApplicationString Condition="'%(CompilerArtifact.NGenApplication)' != ''"> vs.file.ngenApplication="[installDir]\MSBuild\15.0\Bin\Roslyn\%(CompilerArtifact.NGenApplication)"</NGenApplicationString>
       </_File>
-      
+
       <_FileEntries Include='file source="%(_File.Path)"%(_File.NGenArchitectureString)%(_File.NGenPriorityString)%(_File.NGenApplicationString)'/>
+
+      <!-- Satellite assemblies -->
```
Comment (jmarolf) [inline src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj]: Not a big fan of manually specifying each resource assembly that needs to be included.  How will someone know to update this section if/when there are new compiler dependencies?

File: src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj [hunk 2]
```diff
@@ -70,8 +70,68 @@
...
+      <_SatelliteAssembly Include="$(ArtifactsBinDir)Microsoft.CodeAnalysis\$(Configuration)\netstandard2.0\**\Microsoft.CodeAnalysis.resources.dll"/>
```
Comment (jmarolf) [inline src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj]: Does this work?
```XML
<_SatelliteAssembly Include="$(ArtifactsBinDir)Microsoft.CodeAnalysis.CSharp\$(Configuration)\netstandard2.0\**\*.resources.dll"/>
```

That way we at lease pick up all of the resources automatically

Comment (Cosifne) [reply src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Setup.csproj]: Yes, this will work. I will update commit shortly

**Comment by tmeschter** [comment]
I have approved the changes. Can we get a signed build of this, and then produce a validation build of VS with it?

**Comment by Cosifne** [comment]
> 
> 
> I have approved the changes. Can we get a signed build of this, and then produce a validation build of VS with it?

Sure

**Pull request merged**